wildfly-ai/subsystem/src/main/java/org/wildfly/extension/ai/AISubsystemSchema.java

```diff
@@ -45,34 +52,88 @@ public VersionedNamespace<IntVersion, AISubsystemSchema> getNamespace() {
-                        .addChild(factory.builder(InMemoryEmbeddingModelProviderRegistrar.PATH).addAttributes(InMemoryEmbeddingModelProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .build())
-                .addChild(PersistentResourceXMLDescription.decorator("embedding-stores")
-                        .addChild(factory.builder(InMemoryEmbeddingStoreProviderRegistrar.PATH).addAttributes(InMemoryEmbeddingStoreProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .addChild(factory.builder(Neo4jEmbeddingStoreProviderRegistrar.PATH).addAttributes(Neo4jEmbeddingStoreProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .addChild(factory.builder(WeaviateEmbeddingStoreProviderRegistrar.PATH).addAttributes(WeaviateEmbeddingStoreProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .build())
-                .addChild(PersistentResourceXMLDescription.decorator("content-retrievers")
-                        .addChild(factory.builder(EmbeddingStoreContentRetrieverProviderRegistrar.PATH).addAttributes(EmbeddingStoreContentRetrieverProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .addChild(factory.builder(Neo4JContentRetrieverProviderRegistrar.PATH).addAttributes(Neo4JContentRetrieverProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .addChild(factory.builder(WebSearchContentContentRetrieverProviderRegistrar.PATH).addAttributes(WebSearchContentContentRetrieverProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .build())
-                .addChild(PersistentResourceXMLDescription.decorator("mcp")
-                        .addChild(factory.builder(McpToolProviderProviderRegistrar.PATH).addAttributes(McpToolProviderProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .addChild(factory.builder(McpClientSseProviderRegistrar.PATH).addAttributes(McpClientSseProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .addChild(factory.builder(McpClientStdioProviderRegistrar.PATH).addAttributes(McpClientStdioProviderRegistrar.ATTRIBUTES.stream()).build())
-                        .build())
+    public SubsystemResourceRegistrationXMLElement getSubsystemXMLElement() {
+        ResourceXMLChoice providerChoices = this.factory.choice().withCardinality(XMLCardinality.Unbounded.OPTIONAL)
+                .addElement(this.chatLanguageModels())
+                .addElement(this.embeddingModels())
+                .addElement(this.embeddingStores())
+                .addElement(this.contentRetrievers())
+                .addElement(this.mcp())
                 .build();
+        return this.factory.subsystemElement(AISubsystemRegistrar.REGISTRATION)
+                .withContent(providerChoices)
+                .build();
+    }
+
+    private ResourceXMLElement chatLanguageModels() {
+        return this.factory.element(new QName(namespace.getUri(), "chat-language-models"))
```

Comment:
Both SubsystemResourceXMLSchema and ResourceXMLParticleFactory implement QNameResolver, so you can simplify this as:

```java
return this.factory.element(this.resolve("chat-language-models"))